docs/source/tutorials_source/package/tutorial_pretrain_detectron2.py

```diff
@@ -0,0 +1,265 @@
+In this tutorial we use a Faster RCNN with a feature pyramid network (FPN), so make sure you have the relevant file (Base-RCNN-FPN.yaml) in your directory.
+
+
+"""
+
+# %%
+# Imports
+# -------
+#
+# Import the Python frameworks we need for this tutorial.
+import torch
+import lightly
+from detectron2 import config, modeling
+from detectron2.checkpoint import DetectionCheckpointer
+
+# %%
+# Configuration
+# -------------
+# Let's set the configuration parameters for our experiments.
+#
+# We use a batch size of 512 and an input size of 128 in order to fit everything
+# on the available amount of memory on our GPU (16GB). The number of features
+# is set to the default output size of the ResNet50 backbone.
+#
+# We only train for 10 epochs because the focus of this tutorial is on the
+# integration with Detectron2.
+
+num_workers = 8
+batch_size = 512
+input_size = 128
+num_ftrs = 2048
+lr = 0.1
+
+seed = 1
+max_epochs = 10
+
+# use cuda if possible
+device = 'cuda' if torch.cuda.is_available() else 'cpu'
+
+
+# %%
+# You might have downloaded the dataset somewhere else or are using a different one.
+# Set the path to the dataset accordingly. Additionally, make sure to set the
+# path to the config file of the Detectron2 model you want to use.
+# We will be using an RCNN with a feature pyramid network (FPN).
+data_path = '/datasets/freiburg_groceries_dataset/images'
+cfg_path = './Base-RCNN-FPN.yaml'
+
+# %%
+# Initialize the Detectron2 Model
+# --------------------------------
+# 
+# The output of the Detectron2 ResNet50 backbone is a dictionary with the keys
+# `res1` through `res5` (see the `documentation <https://detectron2.readthedocs.io/en/latest/modules/modeling.html#detectron2.modeling.ResNet>`_).
+# The keys correspond to the different stages of the ResNet. In this tutorial, we are only
+# interested in the high-level abstractions from the last layer, `res5`. Therefore,
+# we have to add an additional layer which picks the right output from the dictionary.
+class SelectStage(torch.nn.Module):
+    """Selects features from a given stage."""
+
+    def __init__(self, stage: str = 'res5'):
+        super().__init__()
+        self.stage = stage
+
+    def forward(self, x):
+        return x[self.stage]
+
+# %%
+# Let's load the config file and make some adjustments to ensure smooth training.
+cfg = config.get_cfg()
+cfg.merge_from_file(cfg_path)
+
+# use cuda if possible
+cfg.MODEL.DEVICE = device
+
+# randomly initialize network
+cfg.MODEL.WEIGHTS = ""
+
+# detectron2 uses BGR by default but pytorch/torchvision use RGB
+cfg.INPUT.FORMAT = "RGB"
+
+# %%
+# Next, we can build the Detectron2 model and extract the ResNet50 backbone as
+# follows:
+
+detmodel = modeling.build_model(cfg)
+
+backbone = torch.nn.Sequential(
+    detmodel.backbone.bottom_up,
+    SelectStage('res5'),
+    torch.nn.AdaptiveAvgPool2d(1),
```

Comment:
Why do we need this?

Reply:
Do I get this right? The AdaptiveAvgPool2d takes the bsz x 2048 x 4 x 4 input and averages it to a bsz x 2048 output. Why not do a flatten here or why not take subset? Why do the average pooling?